client/src/arcplanner/lib/model/task.dart

```diff
@@ -15,11 +16,17 @@ class Task {
   String _location;
 
   //constructor
-  Task(this._tid, this._aid, this._title, this._description, this._dueDate, this._location);
+  Task(this._aid, this._title,
+      {description = null, dueDate = null, location = null}) {
+    this._tid = new Uuid().v4();
+    this._description = description;
+    this._dueDate = duedate;
+    this._location = location;
+  }
 
   // Defines a user map.  Helps with moving info betwen the db
   //   and the app
```

Comment:
I have not completed my review yet, but I noticed a few extra spaces in a few comments. One of them is on line 28 in the task.dart file, I believe we only need one hanging space.
```suggestion
  //  and the app
```
The other lines I noticed this was on lines 78 and 88 of arc.dart but that may have been intentional.